### ISSUE TITLE
[Input Date Picker, Input Time Picker] Individual month, day, year individual inputs with masking

### ISSUE THREAD

**Issue by SkyeSeitz** [Esri/calcite-design-system#8624]
**monday.com sync:** #18005780783

### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

Similar to [MUI's date picker ](https://mui.com/x/react-date-pickers/date-picker/), make `calcite-date-picker` and `calcite-time-picker`'s month, day, year entries individual inputs, where arrow keys can be used to adjust value. This would also help avoid the current issue with being able to enter invalid characters.
<img width="128" alt="image" src="https://github.com/Esri/calcite-design-system/assets/29716057/eed154da-5a19-44d4-8712-356391f11edb">

UPPERCASE text should now be lowercase.
<img width="347" height="78" alt="Image" src="https://github.com/user-attachments/assets/4326d06e-4aeb-4789-b00a-c41ee90c9d3f" />

### Acceptance Criteria
- include individual inputs for  month, day, year. 
- upArrow & downArrow can be used to increment the values
- rightArrow & leftArrow can be used to move between month, day, year inputs
- direct entry still possible, but consider not accepting invalid characters
- Initial UPPERCASE text moves to lowercase.
- Input Date Picker masked input should auto-advance on valid entry (similar to issue #12225 for Input Time Picker)

The above criteria should be completed for:
- [ ] `calcite-input-date-picker` 
- [x] `calcite-input-time-picker`

### Relevant Info

Related to refactors and enhancements in epic #6714 and masking from #2709

### Which Component

Input Date Picker, Input Time Picker

### Priority impact

p4 - not time sensitive

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (design)

**Comment by geospatialem** [comment]
Once this effort is picked up for development work, a short check-in can be scheduled with Skye to confirm the design efforts for both components.

**Comment by eriklharper** [comment]
Related: #5639

**Comment by geospatialem** [comment]
Blocked by https://github.com/Esri/calcite-design-system/issues/3455

**Comment by DitwanP** [comment]
I can see this being useful but not sure that this is still needed as I think the same thing can be accomplished now by just tabbing to the individual month, year, and day of the `date-picker`. 

Also the criteria of "upArrow & downArrow can be used to increment the values" would change the current behavior of opening the `date-picker` and design because the way the MUI example does this is having a calendar icon and the end of the input that can be used by the keyboard to open the picker. 

@SkyeSeitz @ashetland Any thoughts on this?

**Comment by SkyeSeitz** [comment]
Thanks for the thoughts, Ditwan! I still see this as a valid enhancement to pursue since typing out a date can often be a much more efficient workflow for keyboard users (and beyond) than tabbing through the Date Picker UI. 

Currently, if tabbing into Input Date Picker, the input is activated for direct entry- masked inputs could just allow for a better direct entry experience without even having to open the Date Picker via the downArrow.

Unlike MUI, clicking into Input Date Picker does in fact trigger Date Picker to open, but I am not sure that is an issue since input behavior is essentially the same regardless: 

<img width="631" alt="image" src="https://github.com/user-attachments/assets/c2f84052-9269-403b-ba1d-3511a181f4ed">

**Comment by eriklharper** [comment]
> I can see this being useful but not sure that this is still needed as I think the same thing can be accomplished now by just tabbing to the individual month, year, and day of the `date-picker`.
> 
> Also the criteria of "upArrow & downArrow can be used to increment the values" would change the current behavior of opening the `date-picker` and design because the way the MUI example does this is having a calendar icon and the end of the input that can be used by the keyboard to open the picker.
> 
> @SkyeSeitz @ashetland Any thoughts on this?

The biggest reason we want to move to individual inputs is to ease localization of the time values.  We can translate time separator values (`hourSuffix`, `minuteSuffix` and `secondSuffix`) and make them static so that the user doesn't have to type them, making it easier to parse the localized number and meridiem values.  I recommend we make the `date-picker` popup only activate when the toggle button is activated.  User should be able to focus into the input, type the value they want, and then blur it to move onto the next input without needing to navigate in and out of the popover, which is redundant to the text input anyway.  If they want the popover input experience, they can click it to open it and then enter it that way.  This is how the native Chromium (and other browsers) implement time and date inputs.

**Comment by macandcheese** [comment]
@geospatialem @jcfranco confirming - will we have standalone Input Date and Input Time components when this lands, correct? 

Our story could be simplified by having input-time and input-date components have a `picker` prop that provides the functionality of input-time-picker and input-date-picker.

This way we don't have three components (input-date, input-date-picker, date-picker - to say nothing of input[type="date"])... 

cc @chezHarper @ashetland @SkyeSeitz @matgalla to continue that conversation.

**Comment by eriklharper** [comment]
> Our story could be simplified by having input-time and input-date components have a picker prop that provides the functionality of input-time-picker and input-date-picker.

The way I'm handling this on #8480 is changing the popup to only show when the open/close chevron is clicked.  And, yes, I believe that the date and time components are still existing separately, this issue is just about changing the input experience to use individual inputs for each component of the value in the input box.

**Comment by jcfranco** [comment]
> we have standalone Input Date and Input Time components when this lands, correct?

If we're talking about the pickers, yes. If we're talking new components `input-date` and `input-time`, no.

> Our story could be simplified by having input-time and input-date components have a picker prop that provides the functionality of input-time-picker and input-date-picker.

I like that direction; definitely worth exploring. We'd have to refactor components to dynamically fetch functionality to make sure components bring only the bits for the appropriate configuration.

> We could continue to support browser / OS picker with a useNativePicker prop or something

I'd like for us to do some research and discuss options before moving forward on this.

> and remove the date value from the "plain" Input component.

This likely depends on the above point, methinks.

**Comment by DitwanP** [comment]
🍠 Verified

**Comment by eriklharper** [comment]
We should make either a new issue or keep this one open to handle this for `calcite-input-date-picker`.
